src/olympia/devhub/forms.py

```diff
@@ -551,10 +551,8 @@ class NewAddonForm(AddonUploadForm):
     def clean(self):
         if not self.errors:
             self._clean_upload()
-            xpi = parse_addon(self.cleaned_data['upload'])
-            # We don't enforce name uniqueness for unlisted add-ons.
-            if not self.cleaned_data.get('is_unlisted', False):
-                clean_addon_name(xpi['name'], addon_type=xpi['type'])
```

Comment:
The problem with existing ones compared to new ones is that they don't go through review again once you change any detail like the add-on name.